File: halley-ecommerce/src/main/java/com/kodat/of/halleyecommerce/category/CategoryService.java
package com.kodat.of.halleyecommerce.category;

import com.kodat.of.halleyecommerce.common.PageResponse;
import com.kodat.of.halleyecommerce.dto.category.CategoryDto;
import com.kodat.of.halleyecommerce.dto.category.CategoryPathDto;
import com.kodat.of.halleyecommerce.dto.category.CategoryTreeDto;
import com.kodat.of.halleyecommerce.dto.category.MainCategoryDto;
import org.springframework.security.core.Authentication;

import java.util.List;

public interface CategoryService {
    CategoryDto addParentCategory(CategoryDto categoryDto, Authentication connectedAdmin);

    CategoryDto addChildCategory(Long parentCategoryId, CategoryDto categoryDto, Authentication connectedAdmin);

    PageResponse<CategoryDto> getAllCategories(int page,int size);

    List<CategoryTreeDto> getCategoryTree();


    CategoryDto updateCategory(Long categoryId, CategoryDto categoryDto, Authentication connectedAdmin);

    void deleteCategory(Long categoryId, Authentication connectedAdmin);

    List<MainCategoryDto> getMainCategories();

    CategoryTreeDto getCategory(String categorySlug);

    List<CategoryPathDto> getCategoryPaths(String categorySlug);
}

File: halley-next/src/services/categoryService.js
import apiClient from "@/services/axiosInstance";

export const getCategories = () => {
    return apiClient.get(`/categories/tree`);
};

const getMainCategories = () => {
    return apiClient.get(`/categories/getMainCategories`);
}

export const getCategory = (categorySlug) => {
    return apiClient.get(`/categories/${categorySlug}`);
}

const getCategoryPaths = (categorySlug) => {
    return apiClient.get(`/categories/category-paths/${categorySlug}`);
}




const categoryService = {
    getCategories,
    getMainCategories,
    getCategory,
    getCategoryPaths
}

export default categoryService;